packages/jest-circus/src/__mocks__/test_utils.js

```diff
@@ -25,7 +26,11 @@ const BABEL_REGISTER_PATH = require.resolve('babel-register');
 ConditionalTest.skipSuiteOnWindows();
 
 export const runTest = (source: string) => {
-  const tmpFilename = path.join(os.tmpdir(), 'circus-test-file.js');
+  const filename = crypto
+    .createHash('md5')
+    .update(source)
+    .digest('hex');
```

Comment:
this is awesome! we need to make a utility function for this :) we use it all over